Objects/dictobject.c

```diff
@@ -6673,25 +6708,45 @@ _PyObject_StoreInstanceAttribute(PyObject *obj, PyDictValues *values,
         }
 #endif
     }
-    PyDictObject *dict = _PyObject_ManagedDictPointer(obj)->dict;
+
     if (ix == DKIX_EMPTY) {
+        int res;
         if (dict == NULL) {
-            dict = make_dict_from_instance_attributes(
-                    interp, keys, values);
+            // Make the dict but don't publish it in the object
+            // so that no one else will see it.
+            dict = materialize_managed_dict_lock_held(obj);
```

Comment:
I am confused by the comment. Doesn't `materialize_managed_dict_lock_held` internally "publish" the dict?

Reply:
Indeed, this is just an outdated comment from a WIP when I was initially thinking that was the way to go.  I'll delete it.